feat(swizzle): ask user preferred language if no language CLI option provided

yixiaojiu opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.
You can learn more about contributing to Docusaurus here: https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md
Happy contributing!
-->

## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).

<!--
Please also remember to sign the CLA, although you can also sign it after submitting the PR. The CLA is required for us to merge your PR.
If this PR adds or changes functionality, please take some time to update the docs. You can also write docs after the API design is finalized and the code changes have been approved.
-->

## Motivation

<!-- Help us understand your motivation by explaining why you decided to make this change. Does this fix a bug? Does it close an issue? -->

Following the issue https://github.com/facebook/docusaurus/issues/9402, I created a prompt that just shows up when neither `--javascript` or `--typescript` is passed

fix https://github.com/facebook/docusaurus/issues/9402

## Usage screenshot

<img width="1375" alt="image" src="https://github.com/facebook/docusaurus/assets/63455218/385c5f86-b437-46aa-94cf-704c2b82c8da">

## Related issues/PRs

<!-- If you haven't already, link to issues/PRs that are related to this change. This helps us develop the context and keep a rich repo history. If this PR is a continuation of a past PR's work, link to that PR. If the PR addresses part of the problem in a meta-issue, mention that issue. -->
https://github.com/facebook/docusaurus/issues/9402

yixiaojiu committed:
feat(swizzle): added a language prompt when no flag is provided

slorber committed:
Merge branch 'main' into feature-language-selector-swizzle

slorber committed:
typo

slorber committed:
ensure swizzle unit tests do not prompt for language

slorber committed:
use the askPreferredLanguage helper from docusaurus utils

slorber commented:
Thanks, LGTM 👍 

I refactored a bit, deduplicating the prompting util extracted in https://github.com/facebook/docusaurus/pull/9442

slorber committed:
Do not prompt the user if theme only supports JS
Fail-fast if user uses --typescript option with a theme that does not support TS
